File: 238-product-of-array-except-self/238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] lprod = new int[n];
        int[] rprod = new int[n];
        lprod[0] = 1;
        for(int i=1;i<n;i++) lprod[i] = lprod[i-1]*nums[i-1];
        rprod[n-1] = 1;
        for(int i=n-2;i>=0;i--) rprod[i] = rprod[i+1]*nums[i+1];
        for(int i=0;i<n;i++) nums[i] = lprod[i]*rprod[i];
        return nums;
    }
}

File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n:int = len(nums)
        lprod:list[int] = [1]*n
        rprod:list[int] = [1]*n
        lprod[0] = 1
        for i in range(1,n):
            lprod[i] = lprod[i-1]*nums[i-1]
        rprod[-1] = 1
        for i in range(-2,-n-1,-1):
            rprod[i] = rprod[i+1]*nums[i+1]
        return [lprod[i]*rprod[i] for i in range(n)]
        